kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/subscription/SubscriptionManager.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.subscription
+
+import android.util.Log
+import org.eclipse.kuksa.DataBrokerApiInteraction
+import org.eclipse.kuksa.DataBrokerException
+import org.eclipse.kuksa.PropertyObserver
+import org.eclipse.kuksa.VssSpecificationObserver
+import org.eclipse.kuksa.extension.TAG
+import org.eclipse.kuksa.proto.v1.Types
+import org.eclipse.kuksa.proto.v1.Types.Field
+import org.eclipse.kuksa.vsscore.model.VssProperty
+import org.eclipse.kuksa.vsscore.model.VssSpecification
+import org.eclipse.kuksa.vsscore.model.heritage
+
+// TODO find better name
```

Comment:
Open for discussion :) I didn't find a better name, or should we make it SubscriptionEngine?

Reply:
Coupled to the Data broker so "DataBrokerSubscriber"? ;) You know my "love" for "manager" :p